### DISCUSSION TITLE
Understanding of Global State and Accent Base Color

### DISCUSSION THREAD

**Discussion by AathifMahir** [microsoft/fluentui-blazor#778] [General]
Hello @vnbaaij

Before working on the Extended Templates, I thought to myself to work on Theme Service First which i stated on previous discussion to Make the Theming on Extended Template Easier but going to through the source, i'm not understaning how theming currently works, For Instance What's the difference between SetColor on Global State and Set Method on AccentBaseColor class.

Would you mind explaning how the theming works?

by the way, initial idea on Theme Service is Link with FluentDesignSystemProvider and Initializing Theme Service from there and with a parameter on FluentSystemDesignProvider to Enable Persistency

**Comment by vnbaaij** [comment]
Hi Aathif,

Whatever you do, do **not** base your work on the `FluentDesignSystemProvider`. That component is only there for legacy purposes but does not support all the design tokens.

Global state is a sort of simple Theme service. There are only a couple of properties in there at the moment. `SetColor` method on the `GlobalState` class calls a `NotifyStateHasChanged` method that you can subscribe to in other parts of the application. It does not actually make the color change. That is what the `Set` method on the `AccentBaseColor` class does

**Comment by AathifMahir** [comment]
> Hi Aathif,
> 
> Whatever you do, do **not** base your work on the `FluentDesignSystemProvider`. That component is only there for legacy purposes but does not support all the design tokens.
> 
> Global state is a sort of simple Theme service. There are only a couple of properties in there at the moment. `SetColor` method on the `GlobalState` class calls a `NotifyStateHasChanged` method that you can subscribe to in other parts of the application. It does not actually make the color change. That is what the `Set` method on the `AccentBaseColor` class does
> 

Thanks for elaborating on this, I'll start working on these shortly
